### PR TITLE
Major changes

### PR DESCRIPTION
**Pull request by Shashank-g1**

### CODE REVIEW

File: src/Common-Components/Loader/Loader.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import Spinner from "react-bootstrap/Spinner";
+import styles from "./Loader.module.less";
+
+function Loader() {
+  return (
+    <>
+      <div className={styles.loader}>
+        <Spinner animation="border" />
+      </div>
+      ;
```
Comment (gaurav702) [inline src/Common-Components/Loader/Loader.js]: remove ;

File: src/Common-Components/Loader/Loader.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import Spinner from "react-bootstrap/Spinner";
+import styles from "./Loader.module.less";
+
+function Loader() {
+  return (
+    <>
```
Comment (gaurav702) [inline src/Common-Components/Loader/Loader.js]: <> is not required here.

File: src/pages/HomePage/HomePage.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  useEffect(() => {
+    fetchingBlogList();
+  }, [fetchingBlogList]);
+
+  return (
+    <>
+      <NavBar />
+
+      {isFetchingBlogList === true ? (
```
Comment (gaurav702) [inline src/pages/HomePage/HomePage.js]: isFetchingBlogList===true is not required only isFetchingBlogLis is enough

File: src/pages/HomePage/HomePage.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        <Loader />
+      ) : (
+        <div className={styles.blog}>
+          <div className={styles.blog__header}>Featured Articles</div>
+          <hr />
+          <div className={styles.blog__description}>
+            {blogList.map((blog) => (
```
Comment (gaurav702) [inline src/pages/HomePage/HomePage.js]: destructure id and title

File: src/pages/HomePage/index.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import { connect } from "react-redux";
 import HomePage from "./HomePage";
-import { displayedBlogList } from "../state/blogActions";
+import { fetchingBlogList } from "../../state/blogActions";
```
Comment (gaurav702) [inline src/pages/HomePage/index.js]: rename it to fetchBlogList

File: src/pages/ShowBlogDetails/ShowBlogDetails.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import React, { useEffect } from "react";
+import { useParams } from "react-router";
+import styles from "./ShowBlogDetails.module.less";
+import Loader from "../../Common-Components/Loader/Loader";
+
+function ShowBlogDetails(props) {
+  const id = useParams();
+
+  const { id: blogId } = id;
```
Comment (gaurav702) [inline src/pages/ShowBlogDetails/ShowBlogDetails.js]: can be destructured in line 7 only like: const { id: blogId } = useParams()

File: src/pages/ShowBlogDetails/ShowBlogDetails.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      {isFetchingBlogData === true ? (
+        <Loader />
+      ) : (
+        <div className={styles.bodyBackground}>
+          <div className={styles.container}>
+            <h2>Hello user {blogId}</h2>
+
+            <div className={styles.description}>
+              <h5>
+                <b>Title :</b> {blogData.title}
```
Comment (gaurav702) [inline src/pages/ShowBlogDetails/ShowBlogDetails.js]: destructure properties from blogData

File: src/Common-Components/Loader/Loader.js (hunk 1 above)
Comment (Shashank-g1) [reply src/Common-Components/Loader/Loader.js]: Done

File: src/Common-Components/Loader/Loader.js (hunk 2 above)
Comment (Shashank-g1) [reply src/Common-Components/Loader/Loader.js]: Resolved

File: src/pages/HomePage/HomePage.js (hunk 3 above)
Comment (Shashank-g1) [reply src/pages/HomePage/HomePage.js]: Resolved

File: src/pages/ShowBlogDetails/ShowBlogDetails.js (hunk 6 above)
Comment (Shashank-g1) [reply src/pages/ShowBlogDetails/ShowBlogDetails.js]: Done

File: src/pages/ShowBlogDetails/ShowBlogDetails.js (hunk 7 above)
Comment (Shashank-g1) [reply src/pages/ShowBlogDetails/ShowBlogDetails.js]: Done

File: src/pages/HomePage/HomePage.js (hunk 4 above)
Comment (Shashank-g1) [reply src/pages/HomePage/HomePage.js]: Done

File: src/pages/HomePage/index.js (hunk 5 above)
Comment (Shashank-g1) [reply src/pages/HomePage/index.js]: Done

**Pull request merged**